src/autocompleters/CountryAutocomplete.ts

```diff
@@ -0,0 +1,75 @@
+import { Autocompleter } from "../Autocompleter";
+import { ApplicationCommandOptionChoice, AutocompleteInteraction } from "discord.js";
+import { countryData, CountryKey, typeOfCountryProperty } from "../util/countryDataManager";
+import { logger } from "../logger";
+
+class CountryAutocompleter extends Autocompleter {
+	constructor() {
+		super("country");
+	}
+
+	async execute(interaction: AutocompleteInteraction): Promise<void> {
+		if (countryData.length === 0) {
+			logger.debug("can't autocomplete because the country data is still being initialized");
+			await interaction.respond([
+				returnChoiceWithSameValues("I can't autocomplete because the data is still being initialized"),
+			]);
+			return;
+		}
+
+		const input = interaction.options.getFocused() as string;
+
+		if (interaction.options.getSubcommand() == "query") {
+			// autocomplete the choices for the filtering
+			const criteria: CountryKey = (interaction.options.getString("filter-criteria") ?? "population") as CountryKey;
+
+			if ((criteria as string) === "none") {
+				await interaction.respond([returnChoiceWithSameValues("none")]);
+				return;
+			}
+
+			const criteriaType: string = typeOfCountryProperty(criteria);
+			if (criteriaType === "number") {
+				if (input !== "") {
+					await interaction.respond([{ name: input, value: input }]);
+				} else {
+					await interaction.respond([
+						{ name: "please input a number", value: "0" },
+						returnChoiceWithSameValues("100e6"),
+						returnChoiceWithSameValues("10e6"),
+						returnChoiceWithSameValues("1e6"),
+						returnChoiceWithSameValues("100000"),
+						returnChoiceWithSameValues("10000"),
+						returnChoiceWithSameValues("1000"),
+						returnChoiceWithSameValues("0"),
+					]);
+				}
+			} else if (criteriaType === "string" || criteriaType === "object") {
+				let options: string[] = countryData.flatMap((c) => c[criteria]) as string[];
+				options = options.filter((t) => t.toLowerCase().startsWith(input.toLowerCase()));
+
+				// removes the duplicates
+				options = [...new Set(options)];
+
+				await interaction.respond(options.map(returnChoiceWithSameValues).slice(0, 25));
+			} else if (criteriaType === "boolean") {
+				await interaction.respond([returnChoiceWithSameValues("true"), returnChoiceWithSameValues("false")]);
+			}
+
+			return;
+		}
+
+		// any other command only needs the default country autocompletion
+		const matchingCountries: string[] = countryData
+			.map((c) => c.name)
+			.filter((t) => t.toLowerCase().startsWith(input.toLowerCase()));
```

Comment:
What about an inclusion check?
For example `korea` would return both north and south korea as options.
```suggestion
			.filter((t) => t.toLowerCase().includes(input.toLowerCase()));
```